### DISCUSSION TITLE
Camera recommendation

### DISCUSSION THREAD

**Discussion by balajeek** [blakeblackshear/frigate#1540] [General]
I started frigate with 3 wyze cameras with rtsp's, always having issues with camera feeds, some it works on wyze app and not on frigate, so i am not sure what is going on.  So i decided to get a new camera

Any good camera recommendation that does rtsp, and no cloud for privacy, and works well with frigate.

**Comment by scooper1** [comment]
Hikvision color vu.  Seams to work well with frigate and works well at night when  camera LEDs are turned off

**Reply by shred86** [reply]
I just ordered an [Amcrest IP8M-T2599EW](https://www.amazon.com/dp/B07RHQGS8V?psc=1&smid=AM1AKWRN957PC&ref_=chk_typ_imgToDp) off Amazon which are on sale for $93.49. I currently have an older system with 4 cameras hooked up to an NVR but I'm looking to just replace the NVR altogether with Frigate. Going to see how this one works before I order 3 more, but I think the big thing to look for besides a quality sensor is H264 compatibility to work with Frigate. Also, it's worth paying attention to what streams are available. For example, this Amcrest does two streams, the main stream at either 3840x2160 or 2688x1520 and the sub stream at 704x576 or 704x480.

**Reply by fcasarsa** [reply]
I have unifi cameras that have 3 streams where it is possible to fully configure FPS and bitrate for each stream. Resolutions are 1920p 1024p and 640p. Cameras ( not all from unifi ) can be used as standalone without any camera controller.

**Reply by hunterjm** [reply]
> I just ordered an [Amcrest IP8M-T2599EW](https://www.amazon.com/dp/B07RHQGS8V?psc=1&smid=AM1AKWRN957PC&ref_=chk_typ_imgToDp) off Amazon

Just a heads up, I did a comparison between that exact model, and the 5MP (2k) model recently at my parents house. We returned the 4k ones and instead bought the much cheaper 5MP ones.

**Reply by shred86** [reply]
I ended up cancelling the Amcrest order before it arrived and went with the Loryta (Dahua) 5442 which is a 4MP camera but has a significantly better sensor. Appreciate the info though and further shows the importance of not falling into the megapixel hype and focusing on the actual camera sensor itself.

**Comment by balajeek** [comment]
anyone used reolink 5MP?

**Reply by balajeek** [reply]
Hikvision features sound cool
which model you have?
Seem like there is no authorized dealer to buy here at North America

**Reply by trythatagain** [reply]
> anyone used reolink 5MP?

Yes, I have a Reolink RLC-420 5MP, works very well. It was my outside rear camera, but replaced it first with a Reolink 820a, but that only has H265, so bought another Amcrest IP8M-T2499EB (rebadged Dahua with different firmware), which is better than the 420 and to me also better than the 820A.

**Comment by blakeblackshear** [comment]
I usually recommend Dahua, Hikvision, and Amcrest in that order. I personally use Dahuas because they are easier to find. I typically order the Loryta branded Dahuas on Amazon. Search for "Loryta 5442" to get the models comparable to the Hikvision Color Vu.

**Reply by shred86** [reply]
I'd be curious to know what the big reasons are for your particular order of recommendations. Image quality or anything particular that makes it work better with Frigate? Or both? I always thought Amcrest were rebranded Dahua. Now you have me thinking about canceling my order for the Loryta :)

**Reply by blakeblackshear** [reply]
In my experience Dahua and Hikvision both have multiple streams with configurable resolutions and frame rates. They are also rock solid streams. Frigate runs for months without a single hiccup in the streams. They also both have models with large sensors well known for excellent image quality at night. Not all the models are equal. Larger sensors are better than higher resolutions.

For me, Dahua edges out Hikvision because they are easier to find and order, not because they are better cameras. 

Amcrest is my fallback recommendation because they are rebranded Dahuas. They are just rebranding the lower end models with smaller sensors or less configuration options.

**Reply by shred86** [reply]
Ah, appreciate the info. I just cancelled my Amcrest order, lol. Back to doing more research. The Loryta 5442 is looking like the one I might go with. I like the fact it has different lens options too since I have one area I'd prefer a different FOV.

**Reply by PrplHaz4** [reply]
The Amcrest firmware seems to be limited compared to Dahua and rarely gets updates.  There have been recent (successful) efforts to load Dahua firmware on the Amcrest cameras, so that may make their pricing more appealing.  This is only true for very specific models at this point, so make sure to do some research if you want to save some money.

I've also found that the 4k resolution leads to more headaches than its worth (resource usage and compatibility) and like Blake said should probably have erred towards the larger sensor at the expense of resolution.

**Reply by shred86** [reply]
Good to know. Yeah, I'd much prefer a better sensor (especially at night). I was likely only going to run at 4MP instead of 8MP anyways based on just storage alone. I think I've narrowed it down to either the Loryta IPC-T5442TM-AS or EmpireTech IPC-T5442T-ZE. I'd prefer the latter based on it being a varifocal lens, but I can't tell if they both have the same sensor performance at night. I'm assuming the Loryta T5542TM-AS does based on the specs showing a min illumination of 0.002 lux at F1.6 versus the EmpireTech T5542T-ZE with a min illumination of 0.003 lux at F1.8.

For anyone else considering these cameras, the big benefits I've found over the Amcrest:
- Much better quality sensor, especially at night performance. The T554TM-AS seems to be one of the best in terms of night image quality, especially after they fixed the SmartIR feature so subjects aren't washed out at close ranges.
- Supports three streams instead of just two. I only need two but it offers the flexibility to go up to a 1080p sub stream versus just being stuck at the lower D1 resolutions.
- All metal casing and build - a lot of the Amcrest reviews mention it's plastic.

**Reply by itkfilelor** [reply]
As an addition, I am running a [hacked](https://github.com/guino/BazzDoorbell) Geeni camera as a nanny cam for my sons room. I hooked it up to frigate because I am raising a 3 y/o ninja. So when I am in my basement den at night, I get a motion alert when the little guy gets to his door trying to steal cookies. :laughing: once configured properly its like a Ronco, set it and forget it.

**Comment by balajeek** [comment]
Great to hear your inputs, I think i will order the 5442, they do have different size 2.8, 3.6 and 6mm at the same price, what different that makes and which would be best?

**Reply by blakeblackshear** [reply]
These are just different lenses with different zoom levels. The FOV is different, so depending on the position, you may want to capture a wider area or a more zoomed in image. They also make a varifocal version (EmpireTech IPC-T5442T-ZE) with an adjustable zoom.

**Reply by shred86** [reply]
I just ordered 3 of the 5442, 1x2.8mm and 2x3.6mm. If you google 2.8mm vs 3.6mm security cameras, there’s some good comparison images and videos to show the difference in depth and field of view. In general, it comes down to a trade off of how wide of FOV do you want (e.g. ability to view a very large area) versus how much detail do you want (e.g. identifying faces, license plates, etc.).

The IPC-T5442T-ZE that blakeblackshear mentioned has a varifocal lens but it does cost a bit more and the night image quality is slightly worse than the fixed lens versions based on the fixed lens versions having a lower fstop (ability to let more light in). The difference seems to be very small as the night performance is still extremely good.

Did you find any 6mm versions in stock anywhere? Amazon (US) only seems to have the 2.8 and 3.6.

**Reply by balajeek** [reply]
I ordered 3.6mm, when i wrote my question about mm i did see 6mm  1 in stock though.
This does not have Wi-Fi i believe so i have to find a way to run ethernet cable to power and connect them.

**Comment by blakeblackshear** [comment]
@balajeek @shred86 Looks like a sale is coming this month: https://ipcamtalk.com/threads/empiretech-6-years-anniversay-promotion-plan.58200/

**Reply by shred86** [reply]
Ah shoot, just ordered some last night on Amazon. Well, it looks like the sale will be on Amazon so maybe I can just get them to price match next week. :) Appreciate the heads up.

**Reply by balajeek** [reply]
Good to know, I only ordered one 5442, may be i will grab the empiretech when it comes on sale.

**Reply by balajeek** [reply]
> @balajeek @shred86 Looks like a sale is coming this month: https://ipcamtalk.com/threads/empiretech-6-years-anniversay-promotion-plan.58200/

Which model is worth? seem like the sale is Wed.  Its on amazon.
Looks like i can send email to them directly for even a lower price right now.

**Reply by shred86** [reply]
Yep, I just messaged Andy directly and purchased three more (2.8mm, 3.6mm and 6mm). ;) Looking forward to completely moving my camera setup to just these cameras and Frigate.

**Reply by balajeek** [reply]
Which model did you buy, did he give even lower price that posted on promotion?

**Reply by shred86** [reply]
IPC-T5442TM-AS (Loryta 5442). It was a couple dollars cheaper per camera than the posted promotion pricing. I really just ordered them from him because I was worried the 6mm version would be sold out on Amazon and I’m also impatient. ;)

**Reply by balajeek** [reply]
yep, good move, that make sense to get it now rather be late or not able to get on sale.

**Comment by shred86** [comment]
Just got a Loryta 5442 (3.6mm) yesterday and so far it’s working fairly well. I’ve updated it with the latest firmware from EmpireTech-Andy, but I do notice one issue. I’m running my main stream at 4MP (2688x1520) with H.264. If I set the bit rate to anything higher than 4096, I can’t seem to view the live stream in Home Assistant. It just tries to load then gives an error symbol that nothing loaded. The weird part is it was working fine last night but just started happening this morning. Anyone else with this camera notice similar issues?

**Reply by shred86** [reply]
I'm also noticing these errors in the logs:

````
[2021-08-21 08:42:36] ffmpeg.back.record_rtmp        ERROR   : av_interleaved_write_frame(): Broken pipe
[2021-08-21 08:42:36] ffmpeg.back.record_rtmp        ERROR   : [flv @ 0x562731c0bf40] Failed to update header with correct duration.
[2021-08-21 08:42:36] ffmpeg.back.record_rtmp        ERROR   : [flv @ 0x562731c0bf40] Failed to update header with correct filesize.
[2021-08-21 08:42:36] ffmpeg.back.record_rtmp        ERROR   : Error writing trailer of rtmp://127.0.0.1/live/back: Broken pipe
[2021-08-21 08:42:36] watchdog.back                  INFO    : Terminating the existing ffmpeg process...
````

Haven't changed the ffmpeg section of my config:

````
ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    - -hwaccel_output_format
    - yuv420p
````

**Comment by balajeek** [comment]
my order from amazon for 5442 should be arriving today, and i will give it a try.  is the procedure for firmware somewhere so can be prepared.

**Reply by shred86** [reply]
I used the latest one provided by EmpireTech-Andy that is [posted here](https://ipcamtalk.com/threads/ipc-t5442t-ze-ipc-t5442tm-as-smd-3-0-smart-ir-latest-new-firmware-from-empiretech.56600/). I just updated it from the cameras web interface.

**Comment by shred86** [comment]
Here's where I ended up setting everything after messing around for a bit with the Loryta 5542. What I'm striving for:

- Ability to view live streams on my phone through HomeKit (via Home Assistant). Don't need high resolution necessarily, just something good enough to see what's going around the house if needed.
- Record 24/7 at the highest quality possible while trying to preserve disk space. This would be used in the event something happened (i.e. identify faces, etc.).
- Minimize CPU usage since I'm running this on my mini server with an Intel Celeron N3450 (6W TDP) and a M.2 Coral EdgeTPU with about 9 other Docker containers.

In the Loryta 5442 camera settings:

**Main Stream**
- Encode Mode: H.265
- Encoding Strategy: General
- Resolution: 2688*1520
- Frame Rate(FPS): 15
- Bit Rate Type: CBR
- Bit Rate: 6144
- I Frame Interval: 30

**Sub Stream**
- Enable: Sub Stream 2
- Encode Mode: H.264
- Resolution: 1280*720
- Frame Rate: 15
- Bit Rate Type: CBR
- Bit Rate: 2048
- I Frame Interval: 30
- SVC: 1(off)

Within Frigate, I have the main stream set for `record` and the sub stream set for `detect` and `rtmp`. Going from a 640x480 to 1280x720 resolution for detect at 5fps only seems to have increased my CPU usage by a couple percent. This is with 4 cameras, 3 of which have detect enabled and the other camera only has detect enabled when we're away (indoor camera). My CPU usage sits around 28%.

Using the H.265 on the main stream for recording helps reduce bandwidth usage across the network and smaller file sizes when storing. I primarily use Safari which supports viewing H.265. It appears you must use H.264 for the detect stream though.

Of note, I only have one Loryta 5442 right now. The other three are cameras are an older rebranded Dahua 3MP camera but I'm still running detect in Frigate using 1280x720 on all the cameras. Planning to replace them with more Loryta 5442s here in about a week. Thanks @blakeblackshear for the recommendation on the cameras.

**Comment by balajeek** [comment]
Just received it, was hard time to set it up, as per the manual they set the default ip as 192.168.1.108 and the only way i could change that ip was to run windows program config.exe, luckily even though i tried on mac thru web ui, i run the config from VM and it took a while to figure it out, now i connected the camera and able to access the web ui and camera all fine.

next step i am going to vlc and open the network stream with 
```
rtsp://admin:mypass@192.168.2.54/cam/realmonitor?channel=1&subtype=0 
or 
rtsp://admin:mypass@192.168.2.54:554/cam/realmonitor?channel=1&subtype=0
```
it throws an error "unable to open".  I did exactly what the manual says.  Any idea?

**Reply by shred86** [reply]
That's the same URL I'm using (first one). Does your password maybe have a ? character in it?

**Reply by balajeek** [reply]
no but have #, does it matter?

**Reply by shred86** [reply]
Yeah, try changing your password to not contain a # or ? symbol.

Edit: Really any of [these special characters](https://secure.n-able.com/webhelp/nc_9-1-0_so_en/content/sa_docs/api_level_integration/api_integration_urlencoding.html) should be encoded if used in a URL.

**Reply by balajeek** [reply]
omg, seriously.
i removed the # from the password and it worked.   I wish i could put all the special chars to make it strong but looks like not.
thanks @shred86.

now i am going to try add this into frigate.

**Comment by balajeek** [comment]
or may be whats helpful will be if you can post your camera config for this 5442, I'll copy and adjust.
thanks.

**Reply by blakeblackshear** [reply]
For reference, here is a snippet from one place I am using 5442s. I setup a read only viewer user rather than use the admin.

The streams are set as follows
- detect: 1080p at 5fps
- clips/record: full resolution at 15fps
- rtmp: 704 (i think) at 10fps. this reduces bandwidth for remote viewing

```yaml
  back:
    ffmpeg:
      inputs: 
        - path: rtsp://viewer:{FRIGATE_CAMERA_PASSWORD}@10.0.10.11:554/cam/realmonitor?channel=1&subtype=2
          roles:
            - detect
        - path: rtsp://viewer:{FRIGATE_CAMERA_PASSWORD}@10.0.10.11:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - rtmp
        - path: rtsp://viewer:{FRIGATE_CAMERA_PASSWORD}@10.0.10.11:554/live
          roles:
            - clips
            - record
    motion:
      mask: 1125,1078,1236,504,444,419,245,622,21,539,4,544,9,488,145,336,255,288,382,225,545,178,730,146,920,137,1112,137,1285,148,1439,175,1602,220,1748,271,1838,348,1900,475,1919,626,1919,7,1495,0,0,2,1,627,9,1077,1,1078
    clips:
      enabled: True
    record:
      enabled: True
    snapshots:
      enabled: True
    width: 1920
    height: 1080
```

**Reply by shred86** [reply]
What’s the reasoning for using a 1080p stream for detect vs something smaller? I’m still trying to wrap my brain around what the docs mention “…choose a camera resolution where the smallest object you want to detect barely fits inside a 300x300px square.”

Doesn’t that depend on the resolution of the stream you’re passing? For example, 300x300px on a 1920x1080 stream vs 640x480 is going to cover a much smaller area.

**Reply by blakeblackshear** [reply]
The objects I want to detect are far enough away that they fit within about a 300x300 pixel area on the 1080p resolution.

**Reply by shred86** [reply]
That makes sense, but wouldn't they still fit in a 300x300px area in a 640x480px stream? I guess the tradeoff is detection performance/accuracy, the smaller the object (in physical size and/or distance) I'm trying to detect, the higher resolution I'd want. For example, if I'm trying to detect a cat across the yard using a 640x480px stream, a cat would probably only be a handful of pixels big making object detection very difficult versus a 1920x1080px stream where the cat would be "higher resolution" making the object detection performance better.

In other words, as the docs state, “…choose a camera resolution where the smallest object you want to detect barely fits inside a 300x300px square.” ;) Makes sense now.

I made a quick graphic to help me visualize the 300x300 area in relation to the common sub stream sizes if it's helpful for anyone else.

![resolutions](https://user-images.githubusercontent.com/32663154/130382173-e82a9503-03ab-4b7f-a34d-d4504b2af82b.jpg)

**Reply by blakeblackshear** [reply]
That's exactly right. You want as good of a 300x300 image as possible. Frigate uses a fast resize method, so the quality drops quick when it has to resize by a significant amount.

**Reply by balajeek** [reply]
Finally got my 50ft network cable to run with POE and just done it this evening.  Now i am trying to set the config like how you did, first i noticed that substream2 is not enabled by default on the camera, so I enabled it but the FPS is showing 1 when 1080p, upto 5fps for 720.  according to your eariler reference you set detect to 1080p at 5fps.
Am I missing something?

**Reply by blakeblackshear** [reply]
You can set the fps and resolutions to whatever works for you. I would also recommend setting your iframe intervals to match the fps.

**Reply by shred86** [reply]
I was reading over at ipcamtalk forums that everyone seems to recommend setting the iframe to double the fps - is it better for Frigate to just match them? I’m not quite sure what it does.

**Reply by balajeek** [reply]
I noticed when set to fps to 5, i see automatically changed to 10 frame rate, and the default mainstream are 30fps and 60 frame rate.

**Reply by blakeblackshear** [reply]
> I was reading over at ipcamtalk forums that everyone seems to recommend setting the iframe to double the fps - is it better for Frigate to just match them? I’m not quite sure what it does.

Double is ok too. That will reduce the bandwidth slightly. Here is an explanation: https://securitycamcenter.com/i-frame-interval-security-camera-systems/

**Comment by balajeek** [comment]
I setup the 5442 camera and replace my front door config on frigate, unfortunately the ethernet cable i had didn't reach far enough so i ordered one from amazon yesterday, until then i pulged in the ethernet over powerline adapter for time being, it very sluggish speed.  all seem work fine so far.  blake thanks for the snippet, I will incorporate this when i connect with ethernet sometime this week.
